### PR TITLE
feat(cli): Add release pipeline (#297)

### LINKED ISSUE #297: CLI: Add release pipeline
## Description

For now, we would like to add a pipeline to deploy the CLI binaries to npm. 

## Solution

Add a workflow file named `deploy-cli.yml`. We only want this to run when a push is made to the `main` branch and any file under `apps/cli`has changed.

## Additional context

If you would like to pick up this issue, we would be really glad! Do join our discord to get in touch with us regarding the npm credentials.

### PR DESCRIPTION
**Pull request by Kiranchaudhary537**
## Description

_Give a summary of the change that you have made_ <br />

Fixes #297 

## Dependencies

_Mention any dependencies/packages used_

## Future Improvements

_Mention any improvements to be done in future related to any file/feature_

## Mentions

_Mention and tag the people_

## Screenshots of relevant screens

_Add screenshots of relevant screens_

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Added a new GitHub Actions workflow (`.github/workflows/deploy-cli.yml`) for deploying the CLI.
- Configured the workflow to trigger on pushes and pull requests to the main branch, specifically targeting changes in the `apps/cli` directory and the workflow file itself.
- Included steps to:
  - Checkout the repository.
  - Set up Node.js version 20.
  - Install dependencies using `pnpm`.
  - Bump the version and generate a changelog using `npm run release`.
  - Publish the package to NPM.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>deploy-cli.yml</strong><dd><code>Add GitHub Actions workflow for CLI deployment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/deploy-cli.yml

<li>Added a new GitHub Actions workflow for deploying the CLI.<br> <li> Configured the workflow to trigger on pushes and pull requests to the <br>main branch.<br> <li> Included steps for checking out the repository, setting up Node.js, <br>installing dependencies, bumping the version, generating a changelog, <br>and publishing to NPM.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/407/files#diff-8b861bd9d09e119c9e80afa2efc4bfd8ba130aed4b7b129deaf4b941701529b7">+39/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: .github/workflows/deploy-cli.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Deploy CLI
+
+on:
+    push:
+      branches:
+        - 'main'
+      paths: ['apps/cli/**','.github/workflows/deploy-cli.yaml']
+    pull_request:
```
Comment (rajdip-b) [inline .github/workflows/deploy-cli.yml:8]: Not on pull requests

File: .github/workflows/deploy-cli.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+  
+      - name: Install dependencies
+        run: pnpm install
+  
+      - name: Bump version and generate changelog
```
Comment (rajdip-b) [inline .github/workflows/deploy-cli.yml:31]: We don't want this step

File: .github/workflows/deploy-cli.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        run: npm run release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  
+      - name: Publish to NPM
```
Comment (rajdip-b) [inline .github/workflows/deploy-cli.yml:36]: We don't have any publish script either. You would need to come up with one that will publish the cli. And lastly, we want to use `pnpm`, not `npm`

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Comment by kriptonian1** [comment]
@Kiranchaudhary537 what's the update man

**Comment by Kiranchaudhary537** [comment]
The requested changes have been implemented, and comments have been refactored. Please let me know if anything is missing.

**Comment by rajdip-b** [comment]
There are a ton of conflicts @Kiranchaudhary537, could you please put up another PR? I believe you haven't made changes to many files.

**Comment by rajdip-b** [comment]
@Kiranchaudhary537 hey man, are you still on it? We would really like to get this merged

**Comment by kriptonian1** [comment]
why is there 472 changed files, you are not supposed to commit everything, you only need to add the files you worked with

**Comment by rajdip-b** [comment]
I agree, you should close this PR, and make your changes on a newer one.

**Comment by rajdip-b** [comment]
Closing due to inactivity.

**Pull request closed without merging**